FirebaseInAppMessaging/Sources/Flows/FIRIAMDisplayExecutor.m

```diff
@@ -209,6 +211,13 @@ - (void)impressionDetectedForMessage:(FIRInAppMessagingDisplayMessage *)inAppMes
     return;
   }
 
+  // If this is an experimental FIAM, activate the experiment.
+  if (inAppMessage.campaignInfo.experimentPayload) {
+    [[FIRExperimentController sharedInstance]
+        activateExperiment:inAppMessage.campaignInfo.experimentPayload
```

Comment:
Just curious why not activate experiments when we get the payload in FIRIAMFetchUsingRestful class.

Reply:
When we get the payload, the messages are cached and we wait for the right time to show the message. We only activate the experiment after the message has been shown (here). Analogous to when an RC experiment gets its activation event fired.